Buildings/Controls/OBC/Utilities/PIDWithAutotuning/Relay/HalfPeriodRatio.mo

```diff
@@ -0,0 +1,182 @@
+                                                         color={255,0,255}));
+  connect(samAddtOntOff.y, tIncrease.u2) annotation (Line(points={{62,30},{64,
+          30},{64,-48},{28,-48},{28,-38},{38,-38}}, color={0,0,127}));
+  connect(gretOntOff.y, samAddtOntOff.trigger) annotation (Line(points={{22,30},
+          {24,30},{24,46},{50,46},{50,42}}, color={255,0,255}));
+  connect(tOnSample.y, maxtOntOff.u1) annotation (Line(points={{-58,70},{-30,70},
+          {-30,76},{-22,76}},
+                           color={0,0,127}));
+  connect(maxtOntOff.u2, tOffSample.y) annotation (Line(points={{-22,64},{-46,64},
+          {-46,-70},{-58,-70}},
+                              color={0,0,127}));
+  connect(mintOntOff.u2, tOffSample.y) annotation (Line(points={{-22,-76},{-46,-76},
+          {-46,-70},{-58,-70}},
+                              color={0,0,127}));
+  connect(mintOntOff.u1, maxtOntOff.u1) annotation (Line(points={{-22,-64},{-54,
+          -64},{-54,70},{-30,70},{-30,76},{-22,76}},
+                            color={0,0,127}));
+  connect(maxtOntOff.y, halPerRat.u1) annotation (Line(points={{2,70},{52,70},{52,
+          76},{58,76}}, color={0,0,127}));
+  connect(halPerRat.u2, mintOntOff.y) annotation (Line(points={{58,64},{32,64},{
+          32,-34},{24,-34},{24,-70},{2,-70}}, color={0,0,127}));
+  connect(halPerRat.y, rho) annotation (Line(points={{82,70},{94,70},{94,60},{120,
+          60}}, color={0,0,127}));
+  connect(triEnd, tOffSample.trigger) annotation (Line(points={{120,-60},{26,-60},
+          {26,-52},{-70,-52},{-70,-58}}, color={255,0,255}));
+  connect(AddtOntOff.u2, tOff) annotation (Line(points={{-82,-36},{-94,-36},{-94,
+          -60},{-120,-60}}, color={0,0,127}));
+  connect(samAddtOntOff.u, tIncrease.u1) annotation (Line(points={{38,30},{28,
+          30},{28,-30},{38,-30}}, color={0,0,127}));
+  connect(tmin.y, mul.u1) annotation (Line(points={{-58,30},{-52,30},{-52,-24},{
+          -42,-24}},            color={0,0,127}));
+  connect(AddtOntOff.u1, tOn) annotation (Line(points={{-82,-24},{-94,-24},{-94,
+          60},{-120,60}},                   color={0,0,127}));
+  connect(AddtOntOff.y, mul.u2) annotation (Line(points={{-58,-30},{-50,-30},{-50,
+          -36},{-42,-36}},                     color={0,0,127}));
+  connect(mul.y, gretOntOff.u1) annotation (Line(points={{-18,-30},{-8,-30},{-8,
+          30},{-2,30}}, color={0,0,127}));
+  connect(mul.y, tIncrease.u1)
+    annotation (Line(points={{-18,-30},{38,-30}}, color={0,0,127}));
+  connect(gretmaxtOntOff.y, triSta)
+    annotation (Line(points={{22,0},{120,0}}, color={255,0,255}));
+  connect(gretmaxtOntOff.u2, gretOntOff.u2) annotation (Line(points={{-2,8},{
+          -16,8},{-16,30},{-10,30},{-10,38},{-2,38}}, color={0,0,127}));
+  connect(gretmaxtOntOff.u1, mul.u2) annotation (Line(points={{-2,0},{-50,0},{
+          -50,-36},{-42,-36}}, color={0,0,127}));
+  connect(tDecrease.u1, tIncrease.u1) annotation (Line(points={{38,-80},{34,-80},
```

Comment:
There are some connections that connect input to input, like in this case, connecting the input `tDecrease.u2` to `tIncrease.u2`. It should connect from an instance output to input of another instance.

Reply:
fixed it and other connects as well